src/gnatss/main.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Dict
+
+import typer
+
+from .configs.main import Configuration
+from .harmonic_mean import sv_harmonic_mean
+from .loaders import load_sound_speed
+from .utilities.io import _get_filesystem
+
+
+def gather_files(config: Configuration) -> Dict[str, Any]:
+    all_files_dict = {}
+    for k, v in config.solver.input_files.dict().items():
+        path = v.get("path", "")
+        typer.echo(f"Gathering {k} at {path}")
+        storage_options = v.get("storage_options", {})
+
+        fs = _get_filesystem(path, storage_options)
+        if "**" in path:
```

Comment:
curious as to why this is needed. Would glob do the wrong thing is the path does **not** contain `**`?

Reply:
It'll give me a list instead of string, so if there's only one file, it'd be `[/path/to/file]` rather than just `/path/to/file`